numalogic/config/_config.py

```diff
@@ -98,6 +99,34 @@ class TrainerConf:
     pltrainer_conf: LightningTrainerConf = field(default_factory=LightningTrainerConf)
 
 
+class AggMethod(str, Enum):
+    EXP = "exp_moving_average"
+    WEIGHTED_AVG = "weighted_average"
+    MEAN = "mean"
+    MAX = "max"
+    MIN = "min"
+
+    @classmethod
+    def get_all(cls) -> list[str]:
+        return [AggMethod.EXP, AggMethod.WEIGHTED_AVG, AggMethod.MEAN, AggMethod.MAX, AggMethod.MIN]
+
+
+@dataclass
+class AggregatorConf:
+    method: AggMethod
+    conf: dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass
+class ScoreConf:
```

Comment:
Should not this come under udf config?

Reply:
Not really, aggregation is needed for the end score. I believe it should come under the base numalogic conf.